### DISCUSSION TITLE
Problems deploying a next app to vercel

### DISCUSSION THREAD

**Discussion by benediktvaldez** [vercel/vercel#4504] [Help] [answered]
I'm having problems deploying this app to vercel.

This is the structure of my folder:
```
┌── .next/
│   ├── cache/...
│   ├── server/...
│   ├── static/...
│   ├── build-manifest.json
│   ├── BUILD_ID
│   ├── export-marker.json
│   ├── prerender-manifest.json
│   ├── react-loadable-manifest.json
│   └── routes-manifest.json
├── .vercel/
│   └── project.json > containing orgId and projectId
├── public/
│   ├── favicon.ico
│   └── robots.txt
├── package.json
└── yarn.lock
```

If I run `next start` within this folder, everything works as expected, locally.

Then I simply run `vercel`, but that results in the following error. As you can see above the routes-manifest is there, in the default output folder `.next`, I tried explicitly setting it as the output folder in settings which didn't change anything.

> Error: A "routes-manifest.json" couldn't be found. Is the correct output directory configured? This setting does not need to be changed in most cases. Learn more: https://err.sh/zeit/now/now-next-routes-manifest

Any thoughts on what I could be doing wrong?

**Comment by styfle** [comment, marked as answer]
You should not need to change the output folder. Ideally, you should keep everything as the default Project Settings.

In the Vercel dashboard, open your "Project Settings" and draw attention to "Build & Development Settings":

1. Ensure that the "Build Command" setting is not changed, or that it calls `next build`. If this command is not changed but you are seeing this error, double check that your `build` script in `package.json` calls `next build`.
2. Ensure that the "Output Directory" setting is not changed. This value almost never needs to be configured, and is only necessary if you override `distDir` in `next.config.js`.
3. For `next export` users: **do not override the "Output Directory"**. Next.js automatically detects what folder you outputted `next export` to.

**Reply by benediktvaldez** [reply]
Yeah none of the settings have been changed, like I mentioned. I'm not using next export, I'm using next build. I'm realising that the problem is that I assumed I could build the app and just push the result.

Is there no way of building the app locally and just pushing the result? The way my setup is (an nx monorepo) it would make things a lot easier.

**Reply by benediktvaldez** [reply]
Sorry for waisting your time, just found this which is literally the same problem I'm having.

https://github.com/vercel/vercel/discussions/4404
